Add monitor sample (#13989)
nrfconnect/sdk-nrf: samples/wifi/monitor/README.rst

.. _wifi_monitor_sample:

> Wi-Fi: Monitor

Reviewer: The way I see it, this sample should be a raw capture sample which can be configured in monitor mode and (later) promiscuous mode . Let's discuss this internally and modify accordingly.

Author: Following internal discussions, maintain "Monitor" as a distinct example, and treat "promiscuous mode" as a separate illustration.

##############

.. contents::
   :local:
   :depth: 2

The Monitor sample demonstrates how to set monitor mode, analyzing incoming Wi-Fi packets, and printing packet statistics.

Requirements
************

The sample supports the following development kits:

.. table-from-sample-yaml::

Overview
********

The sample demonstrates how to configure nrf70 Series device in Monitor mode.
It analyzes the Wi-Fi incoming packets on a raw socket and prints the packet statistics at fixed internal.

> * :kconfig:option:`CONFIG_STATS_PRINT_TIMEOUT`: Wait duration for printing Wi-Fi packet statistics.

Author: Done.

.. _wifi_monitor_sample:

Wi-Fi: Monitor
##############

.. contents::
   :local:
   :depth: 2

> The Monitor sample demonstrates how to set monitor mode, analyzing incoming Wi-Fi packets, and printing packet statistics.

Reviewer 2: ```suggestion
The Monitor sample demonstrates how to set monitor mode, analyze incoming Wi-Fi® packets, and print packet statistics.
```

Reviewer 2: **Wi-Fi®** seems to be missed.

Requirements
************

The sample supports the following development kits:

.. table-from-sample-yaml::

Overview
********

> The sample demonstrates how to configure nrf70 Series device in Monitor mode.

Reviewer 2: ```suggestion
The sample demonstrates how to configure the nRF70 Series device in Monitor mode.
```

Reviewer 2: nRF- **RF** is capitalized.

> It analyzes the Wi-Fi incoming packets on a raw socket and prints the packet statistics at fixed internal.

Reviewer 2: ```suggestion
It analyzes the incoming Wi-Fi packets on a raw socket and prints the packet statistics at a fixed interval.
```

> * :kconfig:option:`CONFIG_STATS_PRINT_TIMEOUT`: Wait duration for printing Wi-Fi packet statistics in seconds.

Reviewer 2: ```suggestion
To set the wait duration for printing Wi-Fi packet statistics in seconds, use the :kconfig:option:`CONFIG_STATS_PRINT_TIMEOUT` Kconfig option.
```

Configuration
*************

|config|

Configuration options
=====================

The following sample-specific Kconfig options are used in this sample (located in :file:`samples/wifi/monitor/Kconfig`):

.. options-from-kconfig::

Building and running
********************

.. |sample path| replace:: :file:`samples/wifi/monitor`

.. include:: /includes/build_and_run_ns.txt

To build for the nRF7002 DK, use the ``nrf7002dk_nrf5340_cpuapp`` build target.

> The following are examples of the CLI commands:
> 
>   .. code-block:: console
> 
>      west build -b nrf7002dk_nrf5340_cpuapp

Reviewer 2: ```suggestion
The following is an example of the CLI command:

.. code-block:: console

   west build -b nrf7002dk_nrf5340_cpuapp
```

      [00:00:03.489,746] <dbg> net_if: rs_timeout: (0x20002e20): RS no respond iface 0x20001448 count 3
      [00:00:05.492,889] <inf> monitor: Management Frames:
      [00:00:05.492,919] <inf> monitor:       Beacon Count: 451
      [00:00:05.492,919] <inf> monitor:       Probe Request Count: 20
      [00:00:05.492,919] <inf> monitor:       Probe Response Count: 194
      [00:00:05.492,919] <inf> monitor: Control Frames:
      [00:00:05.492,919] <inf> monitor:        Ack Count 34
      [00:00:05.492,950] <inf> monitor:        RTS Count 4
      [00:00:05.492,950] <inf> monitor:        CTS Count 82
      [00:00:05.492,950] <inf> monitor:        Block Ack Count 0
      [00:00:05.492,950] <inf> monitor:        Block Ack Req Count 0
      [00:00:05.492,980] <inf> monitor: Data Frames:
      [00:00:05.492,980] <inf> monitor:       Data Count: 5
      [00:00:05.492,980] <inf> monitor:       QoS Data Count: 0
      [00:00:05.492,980] <inf> monitor:       Null Count: 0
      [00:00:05.493,011] <inf> monitor:       QoS Null Count: 0

Dependencies
************

> This sample uses the following library:

Reviewer 2: ```suggestion
This sample uses the following |NCS| library:
```

.. _wifi_monitor_sample:

Wi-Fi: Monitor
##############

.. contents::
   :local:
   :depth: 2

The Monitor sample demonstrates how to set monitor mode, analyze incoming Wi-Fi packets, and print packet statistics.

Requirements
************

The sample supports the following development kits:

.. table-from-sample-yaml::

Overview
********

The sample demonstrates how to configure the nrf70 Series device in Monitor mode.
It analyzes the incoming Wi-Fi packets on a raw socket and prints the packet statistics at a fixed interval.

> To set the wait duration for printing Wi-Fi packet statistics in seconds, use the
> :kconfig:option:`CONFIG_STATS_PRINT_TIMEOUT` Kconfig option.

Reviewer 2: ```suggestion
To set the wait duration for printing Wi-Fi packet statistics in seconds, use the :kconfig:option:`CONFIG_STATS_PRINT_TIMEOUT` Kconfig option.
```
